python/tests/io_tests.py

```diff
@@ -59,6 +59,7 @@ def test_write_to_delta_non_dbr_environment_logging(self):
             ],
         )
 
+    # TODO: FIX ME
```

Comment:
This seems related to my comment above where we need to stop checking the DBR version (or if its a DBR env) and check Delta Versions instead.

Reply:
Yep this test currently fails due to the fact that the OPTIMIZE call runs successfully when correct Delta version is installed.

Reply:
Let's parameterize this test code on the version of the Delta library. For older versions it should expect an Exception, for newer ones, not.

In general I think we should *not* be doing runtime version checking in the library code unless absolutely necessary, but it's fine for test-code when major features are contingent on some dependency version like this.

Reply:
I think that would fix one of the issues that @Sonali-guleria had with merging the streaming work as well

Reply:
It looks like z-order support was added in [Delta 2.0.0](https://github.com/delta-io/delta/releases/tag/v2.0.0). I’ll add a clause to check for this in the test.

Reply:
See f3297a94b5333790ea084f9484e867a0b18b0010. I removed `test_write_to_delta_non_dbr_environment_logging` as it's no longer relevant (z-order is not a Databricks-only feature).